x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elasticsearch/ElasticsearchInternalService.java

```diff
@@ -653,21 +662,24 @@ public void inferRerank(
     ) {
         var request = buildInferenceRequest(model.mlNodeDeploymentId(), new TextSimilarityConfigUpdate(query), inputs, inputType, timeout);
 
-        var modelSettings = (CustomElandRerankTaskSettings) model.getTaskSettings();
-        var requestSettings = CustomElandRerankTaskSettings.fromMap(requestTaskSettings);
-        Boolean returnDocs = CustomElandRerankTaskSettings.of(modelSettings, requestSettings).returnDocuments();
+        var returnDocs = Boolean.TRUE;
+        if (model instanceof CustomElandRerankModel) {
```

Comment:
I was getting a `ClassCastException` here for the default `ElasticRerankerModel` so I'm doing this check. Do we need to support request level task settings for the default model as well?

Reply:
Updated in https://github.com/elastic/elasticsearch/pull/117939/commits/dc16082986b8f251052092c7ace63d291035af30